File: ui_testcafe/tests/e2e/step_definitions/when.ts
import { When } from '@cucumber/cucumber';
import * as home from '../../pages/home';

When(/selects the "([^"]*)" item from the results lists/, async (t, itemPosition: string) => {
  await home.selectItem(itemPosition.toString().replace(/[^0-9]/g, ''));
});

When(/selects a total of "([^"]*)" items/, async (t, itemRandomItems: string) => {
  await home.selectRandomItems(Number(itemRandomItems));
});

When(/goes to shopping cart/, async (t) => {
  await home.goToShoppingCart();
});

File: ui_selenium/features/steps/when.py
from behave import when
from features.pages.home import select_item, select_random_items, go_to_shopping_cart

@when('the user selects the "{item_position}" item from the results lists')
@when('selects the "{item_position}" item from the results lists')
def step_impl(context, item_position: str):
	select_item(context, item_position)

@when('the user selects a total of "{item_random_items}" items')
@when('selects a total of "{itemRandomItems}" items')
def step_impl(context, item_random_items: str):
	select_random_items(context, int(item_random_items))

@when('the user goes to shopping cart')
@when('goes to shopping cart')
def step_impl(context):
	go_to_shopping_cart(context)
